DatadogInternal/Sources/NetworkInstrumentation/DatadogURLSessionHandler.swift

```diff
@@ -8,35 +8,33 @@ import Foundation
 
 /// An interface for processing `URLSession` task interceptions.
 public protocol DatadogURLSessionHandler {
-    /// The interceptor's first party hosts
+    /// The first party hosts configured for this handler.
     var firstPartyHosts: FirstPartyHosts { get }
 
-    /// Tells the interceptor to modify a URL request.
+    /// Modifies the provided request by injecting trace headers.
     ///
     /// - Parameters:
-    ///   - request: The request to intercept.
-    ///   - additionalFirstPartyHosts: Additional 1st-party hosts.
-    /// - Returns: The modified request.
-    func modify(request: URLRequest, headerTypes: Set<TracingHeaderType>) -> URLRequest
+    ///   - request: The request to be modified.
+    ///   - headerTypes: The types of tracing headers to inject into the request.
+    /// - Returns: A tuple containing the modified request and the injected TraceContext. If no trace is injected (e.g., due to sampling),
+    ///            the returned request remains unmodified, and the trace context will be nil.
+    func modify(request: URLRequest, headerTypes: Set<TracingHeaderType>) -> (URLRequest, TraceContext?)
 
-    /// Returns the trace of the current execution context.
-    func traceContext() -> TraceContext?
```

Comment:
💡 No longer needed. It was actually doubling the effort:
- active span was obtained once in `modify(request:)` of Trace handler,
- obtained second time in `intercept(task:)` in `NetworkInstrumentationFeature`,
- it was no-op in RUM handler.

Now we read it only once, in Trace handler and pass the `parentSpanID` in `TraceContext`.